src/toil/jobStores/aws/jobStore.py

```diff
@@ -299,6 +339,73 @@ def deleteFile(self, jobStoreFileID):
         else:
             info.delete()
 
+    nativeUrlRegex = re.compile(r"""
+        # url scheme
+        [sS]3://
+
+        # bucket name
+        #   (1) Must have length 3-255
```

Comment:
The allowed length was actually 3-63, check out http://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html